### DISCUSSION TITLE
Default to `cli.packageManager` when prompting to install

### DISCUSSION THREAD

**Discussion by wSedlacek** [cyrilletuzi/vscode-angular-schematics#454] [Ideas]
It would be nice if any install prompts made by this extension followed the `cli.packageManager` option in `angular.json`

**Comment by cyrilletuzi** [comment]
While the point you mentioned is valid and should be part of a potential feature PR, I'd like to extend the discussion, because the real issue is that this step should not happen at all in a project where Angular CLI is installed. Supporting all package managers would require more than that.

Currently, the extension search packages, and so `@schematics/angular` collections, in a quite simple and more importantly *manual way*, by looking into `node_modules/packagename` or in a parent/ancestor `node_modules`.

There is several reasons to that:

1. the extension could have its own internal `@schematics/angular`, but this solution has several issues:
  - it would need a publish/update of the extension on each Angular patch updates,
  - latest schematics may not work in older Angular versions,
  - unfortunately `@schematics/angular` is not built as a production dependency but as a development dependency, which means it is not self-dependent and has a lot of sub-dependencies (more than 40), which would have many consequences for this extension (bundle size, security risks...).

2. usually a Node program do `require('@schematics/angular')` to find packages, but in the case of a VS Code extension it looks into the *extension* `node_modules`, not inside the *user's project* `node_modules`.

3. Even just for `npm`, I didn't find a lib or an automated solution to find a package *at runtime* and from a custom folder, following the official module resolution of the packager.

The right solution to this issue would be to solve the last point. I think there should be somewhere a tool for that (otherwise how Node and package managers themselves would do it?), but it's hard to find as it's an edge case so not documented.

**Comment by cyrilletuzi** [comment]
Hi @wSedlacek, for information, an extended version of the extension is now available: **Schematics Pro**. You will find all the details on the [dedicated website](https://www.cyrilletuzi.com/schematics-pro/).

Regarding this discussion, it solves 2 things:
- most of the time, the issue is totally avoided as Schematics Pro has its own schematics generator, so it is not dependent on `node_modules` and thus not dependent on the package manager either; it also has the benefit to be faster.
- for the few cases left where the package manager comes into play, Schematics Pro will use the `cli.packageManager` config from `angular.json`.

**Comment by cyrilletuzi** [comment]
Hi @wSedlacek, just to let you know the latest version (v6) of the extension does not rely on `node_modules` anymore.
